packages/synapse-interface/utils/hooks/useStaleQuoteUpdater.ts

```diff
@@ -17,34 +16,42 @@ export const useStaleQuoteUpdater = (
   isWalletPending: boolean,
   staleTimeout: number = 15000 // Default 15_000ms or 15s
 ) => {
+  const [isStale, setIsStale] = useState<boolean>(false)
   const eventListenerRef = useRef<null | (() => void)>(null)
+  const timeoutRef = useRef<null | NodeJS.Timeout>(null)
 
   const quoteTime = quote?.id ? convertUuidToUnix(quote?.id) : null
-  const isValidQuote = isNumber(quoteTime) && !isNull(quoteTime)
+  const isQuoteValid = isNumber(quoteTime) && !isNull(quoteTime)
 
-  const currentTime = useIntervalTimer(staleTimeout, !isValidQuote)
+  useIntervalTimer(staleTimeout, !isQuoteValid)
 
   useEffect(() => {
-    if (isValidQuote && !isQuoteLoading && !isWalletPending) {
-      const timeDifference = calculateTimeBetween(currentTime, quoteTime)
-      const isStaleQuote = timeDifference >= staleTimeout
-
-      if (isStaleQuote) {
-        if (eventListenerRef.current) {
-          document.removeEventListener('mousemove', eventListenerRef.current)
-        }
+    if (isQuoteValid && !isQuoteLoading && !isWalletPending) {
+      timeoutRef.current = setTimeout(() => {
+        eventListenerRef.current = null
+        setIsStale(true)
 
         const newEventListener = () => {
           refreshQuoteCallback()
```

Comment:
One thing that comes to mind: If `refreshQuoteCallback` takes longer than `staleTimeout` to complete, you might end up with multiple queued refresh calls. Does the workflow take that into consideration?

Reply:
At any given point in time, only one event listener will be attached in `eventListenerRef`. Multiple listeners will not queue up since the timeout and listener are cleaned up before the next event listener is set. 

This scenario is covered since the `setTimeout` function that attaches a new event listener will not fire off if a quote is currently loading. Once the `refreshQuoteCallback` completes (one that takes longer than `staleTimeout`), the new quote will trigger the useEffect to restart the workflow to check when to refresh again.

Reply:
Refactored in https://github.com/synapsecns/sanguine/commit/3b7cd9aee20f3e8e735bcdd76b63843ca97aaebd to improve legibility of workflow